yesod-bin/AddHandler.hs

```diff
@@ -214,3 +221,10 @@ mkHandler name pattern methods = unlines
 uncapitalize :: String -> String
 uncapitalize (x:xs) = toLower x : xs
 uncapitalize "" = ""
+
+getSrcDir :: FilePath -> IO FilePath
+getSrcDir cabal = do
+    pd <- flattenPackageDescription <$> readPackageDescription normal cabal
+    let buildInfo = allBuildInfo pd
+        srcDirs = concatMap hsSourceDirs buildInfo
+    return $ head srcDirs
```

Comment:
Interesting, I didn't know that. Nonetheless, it's not promised via the Cabal API, and it could change in the future, causing silent breakage on our end. I'd rather be explicit about that case.